subgraphs/goldfinch/src/mappings/tranched_pool.ts

```diff
@@ -12,118 +13,637 @@ import {
 
 export function handleCreditLineMigrated(event: CreditLineMigrated): void {
-  initOrUpdateTranchedPool(event.address, event.block.timestamp)
-  updatePoolCreditLine(event.address, event.block.timestamp)
+  const market = getOrCreateMarket(event.address.toHexString(), event);
+  market._creditLine = event.params.newCreditLine.toHexString();
+  const creditLineContract = CreditLine.bind(event.params.newCreditLine);
+  const currentLimitResult = creditLineContract.try_currentLimit();
+  if (
+    !currentLimitResult.reverted &&
+    currentLimitResult.value.le(BIGINT_ZERO)
+  ) {
+    creditLineContract.interestApr();
+    market.canBorrowFrom = false;
+  }
+  market.save();
+
+  //
+  initOrUpdateTranchedPool(event.address, event.block.timestamp);
+  updatePoolCreditLine(event.address, event.block.timestamp);
 }
 
 export function handleDepositMade(event: DepositMade): void {
-  handleDeposit(event)
+  const marketID = event.address.toHexString();
+  const amount = event.params.amount;
+  const amountUSD = amount.divDecimal(USDC_DECIMALS);
+  const owner = event.params.owner.toHexString();
+
+  const market = getOrCreateMarket(marketID, event);
+
+  const tranchedPoolContract = TranchedPoolContract.bind(event.address);
+  const configContract = GoldfinchConfigContract.bind(
+    tranchedPoolContract.config()
+  );
+  const rewardTokenAddress = Address.fromString(GFI_ADDRESS);
+  const rewardToken = getOrCreateRewardToken(
+    rewardTokenAddress,
+    RewardTokenType.DEPOSIT
+  );
+  market.rewardTokens = [rewardToken.id];
+
+  if (!market._poolToken) {
+    market._poolToken = configContract
+      .getAddress(BigInt.fromI32(CONFIG_KEYS_ADDRESSES.PoolTokens))
+      .toHexString();
+  }
+
+  const creditLineAddress = tranchedPoolContract.creditLine();
+  const creditLineContract = CreditLineContract.bind(creditLineAddress);
+  if (!market._creditLine) {
+    market._creditLine = creditLineAddress.toHexString();
+  }
+
+  const curretLimitResult = creditLineContract.try_currentLimit();
+  if (!curretLimitResult.reverted && curretLimitResult.value.gt(BIGINT_ZERO)) {
+    market.isActive = true;
+    market.canBorrowFrom = true;
+  }
+
+  market.inputTokenBalance = market.inputTokenBalance.plus(amount);
+  market.cumulativeDepositUSD = market.cumulativeDepositUSD.plus(amountUSD);
+  market.totalDepositBalanceUSD =
+    market.inputTokenBalance.divDecimal(USDC_DECIMALS);
+  market.totalValueLockedUSD = market.totalDepositBalanceUSD;
+  // calculate average daily emission since first deposit
+  if (!market._rewardTimestamp) {
+    market._rewardTimestamp = event.block.timestamp;
+    market._cumulativeRewardAmount = BIGINT_ZERO;
+  }
+  market.save();
+
+  const protocol = getOrCreateProtocol();
+  let marketIDs = protocol._marketIDs!;
+  if (marketIDs.indexOf(market.id) < 0) {
+    marketIDs = marketIDs.concat([market.id]);
+  }
+  let totalDepositBalanceUSD = BIGDECIMAL_ZERO;
+  for (let i = 0; i < protocol._marketIDs!.length; i++) {
```

Comment:
shorten up this line by replacing `protocol._marketIDs!.length` w/ `marketIDs.length`